demo/transient/transient_snes_mpfaof90.F90

```diff
@@ -295,73 +297,53 @@ program main
   CHKERRA(ierr);
 
   ncell = (cEnd-cStart)
-  allocate(blockPerm   (ncell*dim*dim));
-  allocate(residualSat (ncell))
   allocate(liquid_mass (ncell))
   allocate(liquid_sat  (ncell))
   allocate(alpha       (ncell))
   allocate(m           (ncell))
   allocate(index       (ncell))
 
+  ! TODO: We disable the PFlotran stuff for now, because it requires us to
+  ! TODO: directly fiddle with characteristic curves.
   if (pflotran_consistent) then
-     call Permeability_PFLOTRAN(perm);
-     call ResidualSat_PFLOTRAN(resSat)
+    print *, "pflotran_consistent option is temporarily disabled!"
+    stop
+    call Permeability_PFLOTRAN(perm);
+    call ResidualSat_PFLOTRAN(resSat)
   else
      call Permeability(perm);
      call ResidualSaturation(resSat)
   end if
-
-  do c = 1,ncell
-    index(c) = c-1;
-    residualSat(c) = resSat
-    do j = 1,dim*dim
-      blockPerm((c-1)*dim*dim+j) = perm(j)
-    enddo
-  enddo
+  call TDySetConstantTensorPermeability(tdy, perm, ierr)
+  CHKERRA(ierr)
+  call TDySetConstantResidualSaturation(tdy, resSat, ierr)
+  CHKERRA(ierr)
 
   call TDySetWaterDensityType(tdy,WATER_DENSITY_EXPONENTIAL,ierr);
   CHKERRA(ierr)
 
   if (pflotran_consistent) then
-!     call TDySetPorosityFunction(tdy,PorosityFunctionPFLOTRAN,0,ierr);
-     call TDySelectPorosityFunction(tdy,"porosity_pflotran",ierr);
-     CHKERRA(ierr);
-
-     do c = 1,ncell
-        index(c) = c-1;
-        call MaterialPropAlpha_PFLOTRAN(alpha(c))
-        call MaterialPropM_PFLOTRAN(m(c))
-     enddo
-
-     call TDySetCharacteristicCurveVanGenuchtenValuesLocal(tdy,ncell,index,m,alpha,ierr)
-     CHKERRA(ierr);
-
-     call TDySetCharacteristicCurveMualemValuesLocal(tdy,ncell,index,m,ierr)
-     CHKERRA(ierr);
-
-     call TDySetBlockPermeabilityValuesLocal(tdy,ncell,index,blockPerm,ierr);
-     CHKERRA(ierr);
-
-     call TDySetResidualSaturationValuesLocal(tdy,ncell,index,residualSat,ierr);
-     CHKERRA(ierr);
-
+!     call TDySetPorosityFunction(tdy,PorosityFunctionPFLOTRAN,ierr);
+!     CHKERRA(ierr);
+!
+!     do c = 1,ncell
+!        index(c) = c-1;
+!        call MaterialPropAlpha_PFLOTRAN(alpha(c))
+!        call MaterialPropM_PFLOTRAN(m(c))
+!     enddo
+!
+!     call TDySetCharacteristicCurveVanGenuchtenValuesLocal(tdy,ncell,index,m,alpha,ierr)
+!     CHKERRA(ierr);
+!
+!     call TDySetCharacteristicCurveMualemValuesLocal(tdy,ncell,index,m,ierr)
+!     CHKERRA(ierr);
```

Comment:
Is there a new issue to add this functionality back in the future?

Reply:
Yes--see #216